File: leetcode/cpp/3226_Number_of_Bit_Changes_to_Make_Two_Integers_Equal.py
class Solution:
    def minChanges(self, n: int, k: int) -> int:
        ret = 0

        for i in range(32):
            n_bit = (n >> i) & 1
            k_bit = (k >> i) & 1

            if n_bit != k_bit:
                if n_bit != 1:
                    return -1
                ret += 1

        return ret

File: leetcode/cpp/3226_Number_of_Bit_Changes_to_Make_Two_Integers_Equal.cpp
#include <bitset>
class Solution {
public:
	int minChanges(int n, int k) {
		bitset<32> bn(n), bk(k);
		bitset<32> xornk(n^k);

		int ret = 0;
		for (int i = 0; i < 32; ++i) {
			// mismatch
			if (xornk[i] == 1) {
				// k has the 1, not n
				if (bn[i] != 1) return -1;

				++ret;
			}
		}
		return ret;
	}
};
